src/d/actor/d_a_fire.cpp

```diff
@@ -4,112 +4,529 @@
+    if (field_0x8FC != -1 && field_0x8FE != -1) {
+        normal_proc();
+    }
+    eventOrder();
+    checkCol();
+    return TRUE;
 }
 
 /* 00000A70-00000AD8       .text search_wind__8daFire_cFv */
 void daFire_c::search_wind() {
-    /* Nonmatching */
+    fopAc_ac_c* pActor = fopAcM_SearchByName(PROC_WindTag);
+    if (pActor) {
+        field_0x8F0 = fopAcM_GetID(pActor);
+    } else {
+        field_0x8F0 = -1;
+    }
 }
 
 /* 00000AD8-00000E8C       .text ctrlEffect__8daFire_cFv */
 void daFire_c::ctrlEffect() {
-    /* Nonmatching */
+    if (field_0x902 == 0) {
+        return;
+    }
+
+    u8 flag = false;
+
+    search_wind();
+    fopAc_ac_c* wind = fopAcM_SearchByID(field_0x8F0);
+
+    if (field_0x2CC[0].ChkTgHit()) {
+        cCcD_Obj* hit_obj = field_0x2CC[0].GetTgHitObj();
+        if (hit_obj) {
+            if (hit_obj->ChkAtType(AT_TYPE_WIND) && wind != NULL && wind == field_0x2CC[0].GetTgHitAc()) {
+                flag = true;
+            } else {
+                if (hit_obj->ChkAtType(AT_TYPE_WATER) || hit_obj->ChkAtType(AT_TYPE_ICE_ARROW)) {
+                    field_0x8D2 = 0x50;
+                    field_0x8D1 = 0;
+                    flag = true;
+                    execStopNowFire();
+                    dComIfGp_particle_set(dPa_name::ID_COMMON_0463, &current.pos);
+                } else if (hit_obj->ChkAtType(AT_TYPE_WIND) && wind != field_0x2CC[0].GetTgHitAc()) {
+                    field_0x8E0 = *field_0x2CC[0].GetTgRVecP();
+
+                    f32 dist_sq = (field_0x8E0.x * field_0x8E0.x + field_0x8E0.z * field_0x8E0.z) / 1000.0f;
+                    field_0x8E0.y = sqrt(dist_sq);
+                    if (!field_0x8E0.normalizeRS()) {
+                        field_0x8E0.set(0.0f, 1.0f, 0.0f);
+                    }
+                }
+            }
+        }
+
+        field_0x2CC[0].ClrTgHit();
+    } else {
+        cLib_chaseF(&field_0x8E0.x, 0.0f, 0.025f);
+        cLib_chaseF(&field_0x8E0.y, 1.0f, 0.025f);
+        cLib_chaseF(&field_0x8E0.z, 0.0f, 0.025f);
+    }
+
+    if (field_0x8D1 == 0 && field_0x8D2 != 0) {
+        field_0x8D2--;
+        flag = true;
+    }
+
+    if (flag) {
+        execStopFire();
+    } else {
+        execPlayFire();
+        fopAcM_seStart(this, JA_SE_OBJ_BAR_FRAME_BURN, 0);
+    }
+
+    if (field_0x8EC != flag) {
+        if (!flag) {
+            fopAcM_seStart(this, JA_SE_OBJ_BAR_FRAME_ON , 0);
+        } else {
+            fopAcM_seStart(this, JA_SE_OBJ_BAR_FRAME_OFF , 0);
+        }
+    }
+
+    setDirParticle(&field_0x8E0);
+    field_0x8EC = flag;
 }
 
 /* 00000E8C-0000102C       .text demo_proc__8daFire_cFv */
 void daFire_c::demo_proc() {
-    /* Nonmatching */
+    static char* action_table[] = {"WAIT", "ON", "OFF"};
+
+    s32 act_idx = dComIfGp_evmng_getMyActIdx(field_0x8F8, action_table, ARRAY_SIZE(action_table), FALSE, 0);
+    switch (act_idx) {
+        case 0:
+            dComIfGp_evmng_cutEnd(field_0x8F8);
+            return;
```

Comment:
These were probably breaks too, since the switch is the last thing in the function
```suggestion
            break;
```